add_libmagic.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+install_source() {
+    # install from source
+    # https://www.darwinsys.com/file/
+    # https://github.com/file/file/blob/FILE5_45/INSTALL#L51
+    (
+        version="file-5.45" &&
+        tmpfile="$(mktemp)" &&
+        curl -sSLo "${tmpfile}" "https://astron.com/pub/file/${version}.tar.gz" &&
+        tar xvf "${tmpfile}" &&
```

Comment:
Just nitpicking, no blocking issues:
- `-f` (`--fail`) would ensure `curl` fails if the transfer is successful but the server responds with a non-2xx (with `-L`, also non-3xx) status code. In that case, the response probably is not usable anyway.
- This request currently returns the file directly, without redirect. To calm the paranoid, maybe omit `-L`?
- Unless the download is extremely fast and downloaded content is huge, and `curl` risks running OOM waiting for `tar` to unpack the archive (none of this is likely here), the `$tmpfile` handling can be replaced by a pipe. Assuming, of course, that `-o pipefail` remains set.

```suggestion
        curl -fsS "https://astron.com/pub/file/${version}.tar.gz" | \
        tar xv &&
```

Reply:
@estarfoo 

Not a big deal, but if going with the piped `curl` output, personally I'd pair that with an explicit `tar xvf -`. Again, for the paranoid.